canonical-web-and-design/maas-ui
# feat(legacy): remove legacy intro/images pages and set up redirects

## Done

- Removed legacy intro and images pages and set up redirects to the react versions
- Set up redirect from legacy pages to react intro pages if MAAS intro or user intro has not been completed 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Unset `config.completed_intro` and `authUser.completed_intro`
- Try to navigate to any other react or legacy page and check that you get redirected to the MAAS intro
- Complete the MAAS intro and get to the user intro
- Try to navigate to any other react or legacy page and check that you get redirected to the user intro
- Complete the user intro
- Check that clicking Images in the header takes you to the react version of the page
- Check that navigating to `/MAAS/l/images` redirects you to the react images page

## Fixes

Fixes canonical-web-and-design/app-squad#119
Fixes canonical-web-and-design/app-squad#122
Fixes canonical-web-and-design/app-squad#124
